apps/server/src/modules/files-storage/entity/filerecord.entity.ts

```diff
@@ -158,26 +158,70 @@ export class FileRecord extends BaseEntity {
 		this.deletedSince = props.deletedSince;
 	}
 
-	updateSecurityCheckStatus(status: ScanStatus, reason: string): void {
+	public updateSecurityCheckStatus(status: ScanStatus, reason: string): void {
 		this.securityCheck.status = status;
 		this.securityCheck.reason = reason;
 		this.securityCheck.updatedAt = new Date();
 		this.securityCheck.requestToken = undefined;
 	}
 
-	markForDelete(): void {
+	public markForDelete(): void {
 		this.deletedSince = new Date();
 	}
 
-	unmarkForDelete(): void {
+	public unmarkForDelete(): void {
 		this.deletedSince = undefined;
 	}
 
-	setName(name: string): void {
+	public setName(name: string): void {
 		if (name.length === 0) {
 			throw new BadRequestException(ErrorType.FILE_NAME_EMPTY);
 		}
 
 		this.name = name;
 	}
+
+	public hasSameName(name: string): boolean {
+		const hasSameName = this.name === name;
+
+		return hasSameName;
+	}
+
+	public getName(): string {
+		return this.name;
+	}
+
+	public isBlocked(): boolean {
+		const isBlocked = this.securityCheck.status === ScanStatus.BLOCKED;
+
+		return isBlocked;
+	}
+
+	public isPending(): boolean {
+		const isPending = this.securityCheck.status === ScanStatus.PENDING;
+
+		return isPending;
+	}
+
+	public isVerified(): boolean {
+		const isVerified = this.securityCheck.status === ScanStatus.VERIFIED;
+
+		return isVerified;
+	}
+
+	public getParentDescriptions(): { parentType: FileRecordParentType; parentId: EntityId } {
```

Comment:
Ich denke, das sollte `getParentDescription` ohne s heißen. Genauso `getDescription` unten. Das ist ja jeweils 1 Beschreibung.

Reply:
Ich finde description nicht ganz das passende Wording. Ich denke bei description irgendwie an einen description text als string. Vielleicht getAttributes?

Reply:
Dem stimme ich zu. Vielleicht getParentInfo? Die gleiche Frage stellt sich ja auch bei getDescription. Könnte man vielleicht getFileInfo nennen, wobei das schon ziemlich uneindeutig ist. Eine andere Möglichkeit wäre, für jede Prop eine eigene getter-Funktion hinzuzufügen.

Reply:
Info hatte ich als erstes auch drin. Habe mir dann aber gesagt das es eine Beschreibung ähnlich der getDescription ist und es schön wäre ein ähnlcihes wording für ähnliche Fälle zu haben.
Description hatte ich gewählt da wir im Service und in den Params die File Metadaten mit Description benannt hatten, die vom Inhalt identisch sind. ( size, name, mimeType )